hedera-mirror-rosetta/Dockerfile

```
@@ -1,10 +1,12 @@
-FROM golang:1.20.2-bullseye as build
+FROM --platform=$BUILDPLATFORM golang:1.20.2-bullseye as build
```

Shouldn't this be TARGETPLATFORM?
```suggestion
FROM --platform=$TARGETPLATFORM golang:1.20.2-bullseye as build
```

---

it should be `BUILDPLATFORM`, so we can cross-compile arm64 binaries on amd64 platform without emulating an arm64 build container on amd64 build host

---

I'm not sure if this is required for various reasons given Go, GCP, building the image using gradle on Github etc., but when I recently created the new PG15+Citus images using `docker buildx` to build a multi-platform Alpine images, I did not modify the `Dockerfile` itself with platform specific changes.

After setting up the build to use with buildx, I simply specified `--platform linux/arm64,linux/amd64` and pushed and that resulted in a single tag on Docker Hub, with OS/ARCH indicated for both platforms.

---

it's performance optimization, reduces the multi-platform image build time for rosetta in our github workflow from 15m to 7m

---

Ok, nice!